Source/Core/Core/PowerPC/Jit64/Jit_LoadStorePaired.cpp

```diff
@@ -226,113 +140,6 @@ void Jit64::psq_lXX(UGeckoInstruction inst)
 
   gpr.Lock(a, b);
 
-  if (gqrIsConstant && gqrValue == 0)
-  {
-    s32 loadOffset = 0;
-    gpr.BindToRegister(a, true, update);
-    X64Reg addr = gpr.RX(a);
-    if (update && jo.memcheck)
-    {
-      addr = RSCRATCH2;
-      MOV(32, R(addr), gpr.R(a));
-    }
-    if (indexed)
-    {
-      if (update)
-      {
-        ADD(32, R(addr), gpr.R(b));
-      }
-      else
-      {
-        addr = RSCRATCH2;
-        if (a && gpr.R(a).IsSimpleReg() && gpr.R(b).IsSimpleReg())
-        {
-          LEA(32, addr, MRegSum(gpr.RX(a), gpr.RX(b)));
-        }
-        else
-        {
-          MOV(32, R(addr), gpr.R(b));
-          if (a)
-            ADD(32, R(addr), gpr.R(a));
-        }
-      }
-    }
-    else
-    {
-      if (update)
-        ADD(32, R(addr), Imm32(offset));
-      else
-        loadOffset = offset;
-    }
-
-    fpr.Lock(s);
-    if (jo.memcheck)
-    {
-      fpr.StoreFromRegister(s);
-      js.revertFprLoad = s;
-    }
-    fpr.BindToRegister(s, false);
-
-    // Let's mirror the JitAsmCommon code and assume all non-MMU loads go to RAM.
-    if (!jo.memcheck)
-    {
-      if (w)
-      {
-        if (cpu_info.bSSSE3)
-        {
-          MOVD_xmm(XMM0, MComplex(RMEM, addr, SCALE_1, loadOffset));
-          PSHUFB(XMM0, M(pbswapShuffle1x4));
-          UNPCKLPS(XMM0, M(m_one));
-        }
-        else
-        {
-          LoadAndSwap(32, RSCRATCH, MComplex(RMEM, addr, SCALE_1, loadOffset));
-          MOVD_xmm(XMM0, R(RSCRATCH));
-          UNPCKLPS(XMM0, M(m_one));
-        }
-      }
-      else
-      {
-        if (cpu_info.bSSSE3)
-        {
-          MOVQ_xmm(XMM0, MComplex(RMEM, addr, SCALE_1, loadOffset));
-          PSHUFB(XMM0, M(pbswapShuffle2x4));
-        }
-        else
-        {
-          LoadAndSwap(64, RSCRATCH, MComplex(RMEM, addr, SCALE_1, loadOffset));
-          ROL(64, R(RSCRATCH), Imm8(32));
-          MOVQ_xmm(XMM0, R(RSCRATCH));
-        }
-      }
-      CVTPS2PD(fpr.RX(s), R(XMM0));
-    }
-    else
-    {
-      BitSet32 registersInUse = CallerSavedRegistersInUse();
-      registersInUse[fpr.RX(s) << 16] = false;
-      if (update)
-        registersInUse[addr] = true;
-      SafeLoadToReg(RSCRATCH, R(addr), w ? 32 : 64, loadOffset, registersInUse, false);
-      MemoryExceptionCheck();
-      if (w)
-      {
-        MOVD_xmm(XMM0, R(RSCRATCH));
-        UNPCKLPS(XMM0, M(m_one));
-      }
-      else
-      {
-        ROL(64, R(RSCRATCH), Imm8(32));
-        MOVQ_xmm(XMM0, R(RSCRATCH));
-      }
-      CVTPS2PD(fpr.RX(s), R(XMM0));
-      if (update)
-        MOV(32, gpr.R(a), R(addr));
-    }
-    gpr.UnlockAll();
-    fpr.UnlockAll();
-    return;
-  }
```

Comment:
It ends up being a performance wash, but a big win for simplicity. By removing some of the code that was effectively duplicated, it also opens up opportunity to centralize some future optimizations. Those show up in the 'Universal load/store code path' and `MOV_sum` changes in 3565 - we have a single optimized load/store path that everything goes through and re-introduce some of these optimizations in that code path.